client-app/ui-kit/composables/useQuantityValidationSchema.test.ts

```diff
@@ -4,42 +4,80 @@ import { LINE_ITEM_QUANTITY_LIMIT } from "@/core/constants";
 import { mockI18n } from "../test-mocks";
 import { useQuantityValidationSchema } from ".";
 
+type ErrorType = { type: string };
+
 describe("use-quantity-validation-schema", () => {
   mockI18n();
 
   it("quantity is positive", () => {
     const { quantitySchema } = useQuantityValidationSchema({});
 
-    expect(quantitySchema.value.isValidSync(1)).toBeTruthy();
-    expect(quantitySchema.value.isValidSync(0)).toBeFalsy();
-    expect(quantitySchema.value.isValidSync(-1)).toBeFalsy();
+    expect(() => quantitySchema.value.validateSync(1)).not.toThrow();
+
+    try {
+      quantitySchema.value.validateSync(0);
+    } catch (err) {
+      const error = err as ErrorType;
+      expect(error.type).toBe("min");
+    }
+
+    try {
+      quantitySchema.value.validateSync(-1);
+    } catch (err) {
+      const error = err as ErrorType;
+      expect(error.type).toBe("min");
+    }
   });
 
   it("quantity is integer", () => {
     const { quantitySchema } = useQuantityValidationSchema({});
 
-    expect(quantitySchema.value.isValidSync(1)).toBeTruthy();
-    expect(quantitySchema.value.isValidSync(1.5)).toBeFalsy();
+    expect(() => quantitySchema.value.validateSync(1)).not.toThrow();
+
+    try {
+      quantitySchema.value.validateSync(1.5);
+    } catch (err) {
+      const error = err as ErrorType;
+      expect(error.type).toBe("integer");
+    }
   });
 
   it("quantity is less than system limit", () => {
     const { quantitySchema } = useQuantityValidationSchema({
       maxQuantity: ref(Number.MAX_SAFE_INTEGER),
     });
 
-    expect(quantitySchema.value.isValidSync(1)).toBeTruthy();
-    expect(quantitySchema.value.isValidSync(LINE_ITEM_QUANTITY_LIMIT)).toBeTruthy();
-    expect(quantitySchema.value.isValidSync(LINE_ITEM_QUANTITY_LIMIT + 1)).toBeFalsy();
-    expect(quantitySchema.value.isValidSync(Number.MAX_SAFE_INTEGER)).toBeFalsy();
+    expect(() => quantitySchema.value.validateSync(1)).not.toThrow();
+    expect(() => quantitySchema.value.validateSync(LINE_ITEM_QUANTITY_LIMIT)).not.toThrow();
+
+    try {
+      quantitySchema.value.validateSync(LINE_ITEM_QUANTITY_LIMIT + 1);
+    } catch (err) {
+      const error = err as ErrorType;
+      expect(error.type).toBe("max");
+    }
+
+    try {
+      quantitySchema.value.validateSync(Number.MAX_SAFE_INTEGER);
+    } catch (err) {
```

Comment:
make simple
catch (error: ErrorType) {
      expect(error.type).toBe("max");

Reply:
Unfortunately there is no way of setting any other types but `any` and `unknown` for error parameter in the `catch` block:

![image](https://github.com/user-attachments/assets/28b80127-7790-45f3-b796-a71bcefdb0ac)


https://byby.dev/ts-try-catch-error-type